ddtrace/tracer.py

```diff
@@ -542,6 +560,23 @@ def _check_new_process(self):
 
         return new_ctx
 
+    def _log_compat(self, level, msg):
+        """Logs a message for the given level.
+
+        Python 2 will not submit logs to stderr if no handler is configured.
+
+        Instead, something like this will be printed to stderr:
+            No handlers could be found for logger "ddtrace.tracer"
+
+        Since the global tracer is configured on import and it is recommended
+        to import the tracer as early as possible, it will likely be the case
+        that there are no handlers installed yet.
+        """
+        if compat.PY2 and not self.log.handlers:
```

Comment:
if you configure a root log handler does it get passed down to all children loggers?

e.g.

```python
import logging
root = logging.getLogger()
ddtrace = logging.getLogger("ddtrace")
tracer = logging.getLogger("ddtrace.tracer")
```

If we configure a handler on `root` or `ddtrace` will `tracer.handlers` reflect that?

Reply:
Hmm good catch, I'll have to confirm but I'm guessing not. IIRC I wanted to use [`hasHandlers()`](https://docs.python.org/3/library/logging.html#logging.Logger.hasHandlers) which will recursively check upwards in the logger hierarchy but it's Python 3 only 😕 so I had to use `handlers`. We might have to backport it.

Reply:
yeah, it isn't much code, but still :/

https://github.com/python/cpython/blob/ca75fec1ed358f7324272608ca952b2d8226d11a/Lib/logging/__init__.py#L1548-L1568